src/main/java/com/google/sps/servlets/EndHelp.java

```diff
@@ -0,0 +1,94 @@
+package com.google.sps.servlets;
+
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceConfig;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.EntityNotFoundException;
+import com.google.appengine.api.datastore.Key;
+import com.google.appengine.api.datastore.KeyFactory;
+import com.google.appengine.api.datastore.Transaction;
+import com.google.firebase.auth.FirebaseAuth;
+import com.google.firebase.auth.FirebaseAuthException;
+import com.google.firebase.auth.UserRecord;
+import com.google.sps.firebase.FirebaseAppManager;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.ConcurrentModificationException;
+import javax.servlet.ServletConfig;
+import javax.servlet.ServletException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+/** Servlet that creates a new class Datastore. */
+@WebServlet("/end-help")
+public class EndHelp extends HttpServlet {
+  FirebaseAuth authInstance;
+  DatastoreService datastore;
+
+  @Override
+  public void init(ServletConfig config) throws ServletException {
+    try {
+      authInstance = FirebaseAuth.getInstance(FirebaseAppManager.getApp());
+    } catch (IOException e) {
+      throw new ServletException(e);
+    }
+  }
+
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    // Get the input from the form.
+    // navigate to /_ah/admin to view Datastore
+
+    datastore = DatastoreServiceFactory.getDatastoreService();
+    System.setProperty(
+        DatastoreServiceConfig.DATASTORE_EMPTY_LIST_SUPPORT, Boolean.TRUE.toString());
+
+    try {
+      String classCode = request.getParameter("classCode").trim();
+
+      int retries = 10;
+      while (true) {
+        Transaction txn = datastore.beginTransaction();
+        try {
+          // Retrive class entity
+          Key classKey = KeyFactory.stringToKey(classCode);
+          Entity classEntity = datastore.get(txn, classKey);
+
+          // Get uID from uEmail
+          String uEmail = request.getParameter("uEmail");
+          UserRecord userRecord = authInstance.getUserByEmail(uEmail);
+          String uID = userRecord.getUid();
+
+          // Update beingHelped
+          ArrayList<String> beingHelped = (ArrayList) classEntity.getProperty("beingHelped");
```

Comment:
Does the `beingHelped` list keep track of who is helping the student? Also if order of the `beingHelped` list doesn't matter, it may be better to store this information in their own entities.

Reply:
I refactored beingHelped to be an embedded entity with the taID as the property key and the userID as the property value.